frontend/views/containers/proposals/Mincome.vue

```diff
@@ -70,13 +77,47 @@ export default {
       ]
     }
   },
+  computed: {
+    ...mapState([
+      'currentGroupId'
+    ]),
+    ...mapGetters([
+      'groupShouldPropose',
+      'groupSettings'
+    ]),
+    inputSuffix () {
+      return `${currencies[this.groupSettings.incomeCurrency]} ${this.groupSettings.incomeCurrency}`
+    },
+    friendlyIncome () {
+      return `${currencies[this.groupSettings.incomeCurrency]}${this.groupSettings.incomeProvided}`
+    }
+  },
   methods: {
-    submit (form) {
-      console.log(
-        'TODO: Logic to Propose Mincome.',
-        'mincome:', this.form.incomeProvided,
-        'reason:', form.reason
-      )
+    async submit (form) {
+      this.form.response = ''
+      this.form.error = false
+
+      if (this.groupShouldPropose) {
+        return console.log(
+          'TODO: Logic to Propose Mincome.',
+          'mincome:', this.form.incomeProvided,
+          'reason:', form.reason
+        )
+      }
+
+      try {
+        const updatedSettings = await sbp(
+          'gi.contracts/group/updateSettings/create',
+          // to avoid numbers with leading zeros (ex: 01)
+          { incomeProvided: parseInt(this.form.incomeProvided, 10) },
```

Comment:
I recommend a few things here:

1. Doing a project-wide search and renaming `incomeProvided` to `mincomeAmount` (type `number`)
2. Adding a `mincomeCurrency` field as well (type `string`)
3. Reading my other comment under `'gi.contracts/group/updateSettings/process'` for how to validate this on the "backend" as well (i.e. on the receiving side, not just the sending side)

Reply:
1. done.
2. the user can only propose a new income amount, not a different currency, so we don't need to sent it.
3. Done.